sycl/test-e2e/DeviceLib/math_fp64_test.cpp

```diff
@@ -61,6 +61,7 @@ void device_math_test(s::queue &deviceQueue) {
         double subnormal;
         *((uint64_t *)&subnormal) = 0xFFFFFFFFFFFFFULL;
 
+        res_access[i++] = rint(0.9);
         res_access[i++] = cos(0.0);
         res_access[i++] = sin(0.0);
         res_access[i++] = log(1.0);
```

Comment:
```suggestion
        res_access[i++] = log(1.0);
        res_access[i++] = log(1L); // long -> double conversion
```
Another case, which doesn't.

Reply:
It seems like this is trying to use the `long double` variant, which isn't supported on CUDA/HIP, and wasn't working before. But it should be fairly straightforward to add to the header. Which target are you seeing this with?

Reply:
My intention was to use `double` overload. Does it work for smaller integer types (i.e. `char`, `short`, `int`)?

The point of this test is to demonstrate that overload resolution doesn't take into account the implicit conversion. See https://en.cppreference.com/w/cpp/language/overload_resolution.html#Ranking_of_implicit_conversion_sequences.

Reply:
Oh I see, I have a reproducer now, that's a very good catch, I'll investigate further.

I wasn't seeing it in these tests because it's calling `log` on constants, so it would use the wrong overload, but then replace the call with a constant before it becomes a problem. I should probably re-write these maths tests to avoid this issue.